design/FY2022/NFP-Latent_Sizing.md

```diff
@@ -0,0 +1,592 @@
+            CalcZoneSizing(CurOverallSimDay, CtrlZoneNum).DesCoolMassFlow;
+        CalcFinalZoneSizing(CtrlZoneNum).CoolDesDay =
+            CalcZoneSizing(CurOverallSimDay, CtrlZoneNum).CoolDesDay;
+        // other design data removed for brevity
+    } else {
+        CalcFinalZoneSizing(CtrlZoneNum).DesCoolDens = state.dataEnvrn->StdRhoAir;
+        // save design cooling load when the there is design cooling load and the
+        // design cooling volume flow rate is zero, i.e., when
+        // design cooling volume flow rate is set to zero due to cooling supply air
+        // temp greater than zone thermostat temperature
+        if (CalcZoneSizing(CurOverallSimDay, CtrlZoneNum).DesCoolLoad >
+            CalcFinalZoneSizing(CtrlZoneNum).DesCoolLoad) {
+            CalcFinalZoneSizing(CtrlZoneNum).DesCoolLoad =
+                CalcZoneSizing(CurOverallSimDay, CtrlZoneNum).DesCoolLoad;
+            CalcFinalZoneSizing(CtrlZoneNum).CoolDesDay =
+                CalcZoneSizing(CurOverallSimDay, CtrlZoneNum).CoolDesDay;
+            CalcFinalZoneSizing(CtrlZoneNum).CoolLoadSeq =
+                CalcZoneSizing(CurOverallSimDay, CtrlZoneNum).CoolLoadSeq;
+        // other design data removed for brevity
+    }
+
+**EndZoneSizingCalc Sizing Data Processing**
+
+    // candidate EMS calling point to customize CalcFinalZoneSizing
+    bool anyEMSRan;
+    ManageEMS(state, EMSManager::EMSCallFrom::ZoneSizing, anyEMSRan, ObjexxFCL::Optional_int_const());
+
+    // now apply EMS overrides (if any)
+    if (AnyEnergyManagementSystemInModel) {
+        for (CtrlZoneNum = 1; CtrlZoneNum <= NumOfZones; ++CtrlZoneNum) {
+            if (CalcFinalZoneSizing(CtrlZoneNum).EMSOverrideDesHeatMassOn) {
+                if (CalcFinalZoneSizing(CtrlZoneNum).DesHeatMassFlow > 0.0)
+                    CalcFinalZoneSizing(CtrlZoneNum).DesHeatMassFlow =
+                        CalcFinalZoneSizing(CtrlZoneNum).EMSValueDesHeatMassFlow;
+            }
+            // and other EMS overrides
+        }
+    }
+    
+And finally, check for zero zone loads for reporting purposes and then report zone sizing results to the `epluszsz.csv` file.
+
+    for (CtrlZoneNum = 1; CtrlZoneNum <= state.dataGlobal->NumOfZones; ++CtrlZoneNum) {
+        if (!ZoneEquipConfig(CtrlZoneNum).IsControlled) continue;
+        if (std::abs(CalcFinalZoneSizing(CtrlZoneNum).DesCoolLoad) <= 1.e-8) {
+            ShowWarningError(state, "Calculated design cooling load for zone=" + CalcFinalZoneSizing(CtrlZoneNum).ZoneName + " is zero.");
+            ShowContinueError(state, "Check Sizing:Zone and ZoneControl:Thermostat inputs.");
+        }
+        if (std::abs(CalcFinalZoneSizing(CtrlZoneNum).DesHeatLoad) <= 1.e-8) {
+            ShowWarningError(state, "Calculated design heating load for zone=" + CalcFinalZoneSizing(CtrlZoneNum).ZoneName + " is zero.");
+            ShowContinueError(state, "Check Sizing:Zone and ZoneControl:Thermostat inputs.");
+        }
+    }
+
```

Comment:
One thing not described in the NFP is how to apply the sensible and latent air flow rates to size components (i.e., how to choose the max of sensible and latent air volume flow rate). Either at this point in code or previously where the time series data is processed.

One thing I thought of is to set CalcZoneSizing(CurOverallSimDay, CtrlZoneNum).DesCoolVolFlow to the max of either flow rate when latent sizing is requested (or maybe in just the FinalZoneSizing array). This method would hopefully size the component to "be able to" meet the latent load without changing other routines such as sizing.

Another possibility is to make that decision in the component sizing module (CoolingCapacitySizing).

The air system would sum the zone air flows the way it does now except will now include logic to include the latent sized air flow rate.

Reply:
Depending on the application, the user might want to size a given system on latent-only, sensible-only, or the max of both requirements. Perhaps the sizing options should include a Maximum option?

Reply:
Might as well. I'll change this to Sensible Load, Latent Load, Sensible And Latent or whatever other names we land on. I don't need the Sensible Load No Latent since the sensible load is nearly the same when the zone RH is held constant. I want to see diffs before, and if, I remove that option. I also think I need a delta w to be similar to SA temperature sizing options. And a humidification set point input field. All this after I got this to run as intended and noticed some missing information.

    Sizing:Zone, (end of object for time being while testing things out)
    A10,\field Zone Load Sizing Method
        \type choice
        \key Sensible Load Only
        \key Sensible And Latent Load
        \key Sensible Load Only No Latent Load
        \default Sensible Load Only No Latent Load
    N21; \field Zone RH Setpoint
        \type real
        \default 50.0

I also think I need a delta w to be similar to SA cooling/heating temperature sizing options. I actually like the delta T input because if the Tstat temp changes during the day, the supply air temp follows that change (which it would do in real applications). And a humidification set point input field (I figured out a way to control zone humidity during sizing without needing a humidistat). And both of these changed to schedules. All this after I got this to run as intended and noticed some missing information. So several more inputs needed for humidification and dehumidification.

    Sizing:Zone,      (same for heating)
    A2, \field Zone Cooling Design Supply Air Temperature Input Method
      \type choice
      \key SupplyAirTemperature
      \key TemperatureDifference
      \default SupplyAirTemperature
    N1, \field Zone Cooling Design Supply Air Temperature
    N2, \field Zone Cooling Design Supply Air Temperature Difference